src/markey/machine.py

```diff
@@ -113,6 +114,23 @@ def parse_arguments(stream, end_token):
         else:
             break
     for keyword in keywords:
-        args.append(keyword)
+        args.append((keyword, 'func_kwarg'))
 
     return tuple(args), kwargs
+
+
+def parse_arguments(stream, end_token):
+    """
+    Helper function for function argument parsing.  Pass it a
+    `TokenStream` and the delimiter token for the argument section and
+    it will extract all position and keyword arguments.
+
+    Returns a ``(args, kwargs)`` tuple.
+    """
+    args, kwargs = parse_arguments_with_type(stream, end_token)
+
+    raw_args = []
+    for arg, type in args:
```

Comment:
This isn't very performant, and is the wrong way of doing code reuse. I think the better route is to modify the original implementation (L89-110) in a helper method, that takes a default-false arg `extract_types`. So then you have a function like this:
```python
def parse_arguments_helper(stream, end_token, extract_types=False):
    # code here
    # L104 code:
    if extract_types:
        args.append((value, 'func_string_arg'))
    else:
        args.append(value)
    # continue rest of code, flipping what is appended to args based on extract_types value
``` 

Now, define your two main functions very cleanly:
```python
def parse_arguments(stream, end_token):
    return parse_arguments_helper(stream, end_token)


def parse_arguments_with_type(stream, end_token):
    return parse_arguments_helper(stream, end_token, extract_types=True)
```